client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -138,13 +140,41 @@ public void subscribe(final SingleEmitter<HttpResponse> responseEmitter) {
                         @Override
                         public void operationComplete(Future<? super Channel> cf) {
                             if (!cf.isSuccess()) {
-                                responseEmitter.onError(cf.cause());
+                                if (!responseEmitter.isDisposed()) {
```

Comment:
Having these checks everywhere makes me wonder if we should wrap responseEmitter with an object that will check `isDisposed()` before it makes any call. This just seems like it will be error prone over time.

Reply:
Just something to think about.

Reply:
Yeah, that's a good idea.